diff --git a/Engine/Source/Runtime/Engine/Classes/Sound/ReverbEffect.h b/Engine/Source/Runtime/Engine/Classes/Sound/ReverbEffect.h
index a2436d270..55dc3bce5 100644
--- a/Engine/Source/Runtime/Engine/Classes/Sound/ReverbEffect.h
+++ b/Engine/Source/Runtime/Engine/Classes/Sound/ReverbEffect.h
@@ -79,7 +79,7 @@ class UReverbEffect : public UObject
 
 protected:
 
-#if WITH_EDITORONLY_DATA
+#if WITH_EDITOR
 	virtual void PostEditChangeProperty(FPropertyChangedEvent& PropertyChangedEvent) override;
 #endif
 };
