diff --git a/Unreal/CarlaUnreal/Config/DefaultInput.ini b/Unreal/CarlaUnreal/Config/DefaultInput.ini
index 1c5a096..57e0072 100644
--- a/Unreal/CarlaUnreal/Config/DefaultInput.ini
+++ b/Unreal/CarlaUnreal/Config/DefaultInput.ini
@@ -113,8 +113,8 @@ DoubleClickTime=0.200000
 +AxisMappings=(AxisName="MoveForward",Scale=-1.000000,Key=S)
 +AxisMappings=(AxisName="MoveUp",Scale=1.000000,Key=E)
 +AxisMappings=(AxisName="MoveUp",Scale=-1.000000,Key=Q)
-DefaultPlayerInputClass=/Script/EnhancedInput.EnhancedPlayerInput
-DefaultInputComponentClass=/Script/EnhancedInput.EnhancedInputComponent
+DefaultPlayerInputClass=/Script/Engine.PlayerInput
+DefaultInputComponentClass=/Script/Engine.InputComponent
 DefaultTouchInterface=/Engine/MobileResources/HUD/DefaultVirtualJoysticks.DefaultVirtualJoysticks
 -ConsoleKeys=Tilde
 +ConsoleKeys=Tilde
