diff --git a/Engine/Source/ThirdParty/ShaderConductor/ShaderConductor/Source/CMakeLists.txt b/Engine/Source/ThirdParty/ShaderConductor/ShaderConductor/Source/CMakeLists.txt
--- a/Engine/Source/ThirdParty/ShaderConductor/ShaderConductor/Source/CMakeLists.txt
+++ b/Engine/Source/ThirdParty/ShaderConductor/ShaderConductor/Source/CMakeLists.txt
@@ -54,7 +54,7 @@ else()
 	foreach(flagVar
 		CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 		set(${flagVar} "${${flagVar}} -W -Wall -Werror -Wno-deprecated")
-		if(NOT (ANDROID OR IOS))
+		if(NOT (ANDROID OR IOS) AND SC_ARCH_NAME MATCHES "^(x86|x64)$")
 			set(${flagVar} "${${flagVar}} -msse2")
 		endif()
 	endforeach()
@@ -94,7 +94,7 @@ else()
 				set(CMAKE_RC_FLAGS "${CMAKE_RC_FLAGS} --target=elf64-x86-64")
 			endif()
 		endif()
-	else()
+	elseif(SC_ARCH_NAME STREQUAL "x86")
 		foreach(flagVar
 			CMAKE_C_FLAGS CMAKE_CXX_FLAGS)
 			set(${flagVar} "${${flagVar}} -m32")
